tensorflow/compiler/tf2xla/kernels/bincount_op.cc

```diff
@@ -0,0 +1,148 @@
+    auto output_rank = output_shape_param.rank();
+    OP_REQUIRES(ctx, output_rank == 0,
+                errors::InvalidArgument("Shape must be rank 0 but is rank ",
+                                        output_rank));
+    OP_REQUIRES_OK(ctx, ctx->ConstantInputAsIntScalar("size", &output_size));
+    OP_REQUIRES(ctx, output_size >= 0,
+                errors::InvalidArgument("size (", output_size,
+                                        ") must be non-negative"));
+    xla::XlaOp idx, updates, output;
+    xla::XlaOp input = ctx->Input(0);
+    auto input_xla_type = ctx->input_xla_type(0);
+    xla::PrimitiveType dtype = ctx->InputXlaType("weights");
+    auto zero = xla::Zero(ctx->builder(), dtype);
+    auto one = xla::One(ctx->builder(), dtype);
+    StatusOr<xla::Shape> input_shape_or = ctx->builder()->GetShape(input);
+    OP_REQUIRES_OK(ctx, input_shape_or.status());
+    auto input_shape = input_shape_or.ValueOrDie();
+    auto size = input_shape.dimensions(0);
+
+    if (!size) {
+      output = xla::Broadcast(zero, {output_size});
+      ctx->SetOutput(0, output);
+      return;
+    }
+    auto rank = input_shape.rank();
+
+    OP_REQUIRES(ctx, rank <= 2,
+                errors::InvalidArgument(
+                    "Shape must be at most rank 2 but is rank ", rank));
+
+    xla::XlaOp weights = ctx->Input(2);
+    StatusOr<xla::Shape> weights_shape_or = ctx->builder()->GetShape(weights);
+    OP_REQUIRES_OK(ctx, weights_shape_or.status());
+
+    auto weights_shape = weights_shape_or.ValueOrDie();
+    auto weights_size = weights_shape.dimensions(0);
+    bool has_weights = false;
+    if (weights_size) {
+      has_weights = true;
+    }
+    xla::Shape output_shape = xla::ShapeUtil::MakeShape(dtype, {output_size});
+    xla::ScatterDimensionNumbers scatter_dnums;
+    scatter_dnums.set_index_vector_dim(1);
+    scatter_dnums.add_inserted_window_dims(0);
+    scatter_dnums.add_scatter_dims_to_operand_dims(0);
+
+    if (rank == 2) {
+      output_shape = xla::ShapeUtil::MakeShape(dtype, {size, output_size});
+      scatter_dnums.add_inserted_window_dims(1);
+      scatter_dnums.add_scatter_dims_to_operand_dims(1);
+      auto i_shape =
+          xla::ShapeUtil::MakeShape(input_xla_type, {input_shape.dimensions()});
+      auto i = xla::Iota(ctx->builder(), i_shape, 0);
+      i = xla::Reshape(
+          i, {input_shape.dimensions(0) * input_shape.dimensions(1), 1});
+      auto j = xla::Reshape(
+          input, {input_shape.dimensions(0) * input_shape.dimensions(1), 1});
+      std::vector<xla::XlaOp> iotas_to_concat;
+      iotas_to_concat.push_back(i);
+      iotas_to_concat.push_back(j);
+      idx = xla::ConcatInDim(ctx->builder(), iotas_to_concat, 1);
+      updates = xla::Broadcast(
+          one, {input_shape.dimensions(0) * input_shape.dimensions(1)});
+      output = xla::Broadcast(
+          zero, {output_shape.dimensions(0), output_shape.dimensions(1)});
+      if (has_weights and !binary_output_) {
```

Comment:
Please use `&&` instead of `and`.

Reply:
Done.
@cheshire  In the case it was requested by clang-format internally we need to align the conf in OSS

Reply:
I don't think It Is part of the Google c++ style guide.

We need to enforce this with compiler flag `-fno-operator-names` /cc @angerson